src/main/java/com/cloudant/fdblucene/FDBIndexOutput.java

```diff
@@ -160,4 +167,23 @@ private void flushTxnBufferIfFull() {
         }
     }
 
+    private void setFileLength(final TransactionContext txc, final long length) {
+        txc.run(txn -> {
+            applyIfExists(txn, value -> {
+                final FileMetaData meta = new FileMetaData(value).setFileLength(length);
+                txn.set(metaKey, meta.pack());
+            });
+            return null;
+        });
+    }
+
+    private void applyIfExists(final Transaction txn, final Consumer<byte[]> fun) {
```

Comment:
Is there a way to do this without a Consumer? The lambda nesting makes it slightly hard to read.

Reply:
heh, I added that to aid clarity (and I expected to use it more than once). I can fold it in, I think.

Reply:
ah, I do in fact call it from two places, so I think it has to stay.